ui/packages/catalog-ui-search/src/main/webapp/react-component/container/workspace-interactions/workspace-interactions.tsx

```diff
@@ -139,6 +141,30 @@ class WorkspaceInteractions extends React.Component<Props, State> {
       wait: true,
     })
   }
+  deletionPrompt = () => {
+    const workspace = store.getWorkspaceById(this.props.workspace.id)
+    const security = new Security(Restrictions.from(workspace))
+
+    if (security.isShared()) {
+      var self = this
+      this.props.listenTo(
+        ConfirmationView.generateConfirmation({
+          prompt:
+            'Are you sure you want to delete this workspace? It has been shared with other users.',
+          no: 'Cancel',
+          yes: 'Delete',
+        }),
+        'change:choice',
+        function(confirmation: any) {
+          if (confirmation.get('choice')) {
+            self.deleteWorkspace()
+          }
+        }.bind(this)
```

Comment:
Consider binding this function in the constructor and then you don't have to bind the function directly here as you use it.

Reply:
By binding this function here don't we limit the scope of its usage? If it was to be reused in other areas it should be done the way you recommended.